transports/ftp/src/test/java/org/mule/transport/ftp/FtpConnectionSocketTimeoutTestCase.java

```diff
@@ -0,0 +1,215 @@
+        setStartContext(false);
+        this.ftplet = ftpLet;
+        this.nameScenario = nameScenario;
+    }
+
+    @Override
+    protected void doSetUp() throws Exception
+    {
+        receiverException = null;
+        latch = new Latch();
+        serverLatch = new Latch();
+        super.doSetUp();
+    }
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> data()
+    {
+        List<Object[]> parameters = new ArrayList<>();
+        parameters.add(new Object[] { ftpLetOnConnectSleep, "Connection Scenario"});
+        parameters.add(new Object[] { ftpLetOnCommandSleep, "Connection Commands Scenario"});
+        parameters.add(new Object[] { ftpLetOnNoOpCommandSleep, "NOOP Commands Scenario"});
+        return parameters;
+    }
+
+    @Test
+    public void testDispatcherTimeoutConnection() throws Exception
+    {
+        muleContext.start();
+        MuleClient client = muleContext.getClient();
+        MuleMessage testMessage = new DefaultMuleMessage("test", new HashMap(), muleContext);
+        MuleMessage result = client.send("vm://in", testMessage, new SimpleOptions(WAIT_TIMEOUT));
+        assertThat(result, is(notNullValue()));
+        assertException(result.getExceptionPayload().getRootException());
+    }
+
+    @Test
+    public void testReceiverTimeout() throws Exception
+    {
+        muleContext.start();
+        latch.await();
+        assertException(getRootCause(receiverException));
+    }
+
+    @After
+    public void tearDown() throws Exception
+    {
+        serverLatch.countDown();
+    }
+
+    private void assertException(Throwable exception)
+    {
+        assertThat("An timeout exception should be triggered in the " + nameScenario, exception, is(notNullValue()));
+        assertThat("SocketTimeoutException should be triggered in the " + nameScenario, exception.getMessage(), containsString(ERROR_MESSAGE));
+    }
+
+    private static Ftplet ftpLetOnCommandSleep = new DefaultFtplet()
+    {
+        @Override
+        public FtpletResult beforeCommand(FtpSession session, FtpRequest request) throws FtpException, IOException
+        {
+            sleep();
+            return null;
+        }
+    };
+
+
+    private static Ftplet ftpLetOnNoOpCommandSleep = new DefaultFtplet()
+    {
+        @Override
+        public FtpletResult beforeCommand(FtpSession session, FtpRequest request) throws FtpException, IOException
+        {
+            if (NOOP_COMMAND.equals(request.getCommand()))
+            {
+                sleep();
+            }
+            return null;
+        }
+    };
+
+    private static Ftplet ftpLetOnConnectSleep = new DefaultFtplet()
+    {
+        @Override
+        public FtpletResult onConnect(FtpSession session) throws FtpException, IOException
+        {
+            sleep();
+            return null;
+        }
+    };
+
+    private static void sleep()
+    {
+        try
+        {
+            serverLatch.await(SERVER_SLEEP, TimeUnit.MILLISECONDS);
```

Comment:
You must assert against the result of the await to be sure that the timeout was reached.